security/cmd/istio_ca/main.go

```diff
@@ -61,6 +61,11 @@ const (
 
 	// The key for the environment variable that specifies the namespace.
 	listenedNamespaceKey = "NAMESPACE"
+
+	// ServiceAccount/DNS pair for generating DNS names in certificates.
+	// TODO: move it to a configmap later when we have more services to support.
+	sidecarInjectorSvcAccount = "istio-sidecar-injector-service-account"
+	sidecarInjectorSvc        = "istio-sidecar-injector"
```

Comment:
Should this be fully qualified, e.g. `istio-sidecar-injector.istio-system.svc`?

Reply:
It's automatically done here:
https://github.com/istio/istio/pull/4216/files#diff-06f11c93a782fce8e0e35ebdb109588aR264
The namespace should always be the same as the service account. So we don't need/want the user to specify the namespace.

Reply:
Actually you raised a good point. We are missing the namespace for the service account.
So if there's the same service account in a random namespace, it will accidentally get the svc in the cert as well.